### PR TITLE
Add execvpe support, conditional on platform

### LINKED ISSUE #682: Add execvpe support
This was removed in #681 because it was broken. Re-adding it is basically undoing fcc22f22a680233dbeb82d230e35fd54cb6de176 (without the feature support and instead conditionally including it on supported platforms). `libc` has `execvpe` included for `haiku`, `linux`, and `openbsd` platforms. If you are a new contributor, go ahead and read `CONTRIBUTING.md`, assign this to yourself, and file a PR when you've gotten somewhere (start the title with `WIP` if it's not ready to merge, and include "closes #682" in the text of the PR).

### PR DESCRIPTION
**Pull request by F1rst-Unicorn**
This closes #682

### CODE REVIEW

File: src/unistd.rs [hunk 1]
```diff
@@ -701,6 +701,28 @@ pub fn execvp(filename: &CString, args: &[CString]) -> Result<Void> {
     Err(Error::Sys(Errno::last()))
 }
 
+/// Replace the current process image with a new one and replicate shell `PATH`
+/// searching behavior (see
+/// [execvpe(3)](http://man7.org/linux/man-pages/man3/exec.3.html)).
```
Comment (asomers) [inline src/unistd.rs]: The `execvpe(3)` should be in backticks.

File: src/unistd.rs [hunk 2]
```diff
@@ -701,6 +701,28 @@ pub fn execvp(filename: &CString, args: &[CString]) -> Result<Void> {
...
+///
+/// This is a combination of `::nix::unistd::execvp` and `::nix::unistd::execve`
+/// to pass an environment and have a search path. See these two for additional
+/// information.
+#[cfg(any(target_os = "haiku",
+          target_os = "linux",
+          target_os = "openbsd"))]
+#[inline]
```
Comment (asomers) [inline src/unistd.rs]: `#[inline]` is not necessary.  The compiler can figure that out on its own if the application is compiled with LTO.

File: src/unistd.rs [hunk 3]
```diff
@@ -701,6 +701,28 @@ pub fn execvp(filename: &CString, args: &[CString]) -> Result<Void> {
     Err(Error::Sys(Errno::last()))
 }
 
+/// Replace the current process image with a new one and replicate shell `PATH`
+/// searching behavior (see
+/// [execvpe(3)](http://man7.org/linux/man-pages/man3/exec.3.html)).
+///
+/// This is a combination of `::nix::unistd::execvp` and `::nix::unistd::execve`
```
Comment (asomers) [inline src/unistd.rs]: This wording suggests that nix itself is implementing the combination of features, when it's actually the OS.  I suggest this wording: "This functions like a combination of `execvp(2)` and `execve(2)` ..."

**Comment by F1rst-Unicorn** [comment]
I adapted your change requests (not sure whether I shall also "Resolve conversation"...)

File: src/unistd.rs (hunk 3 above)
Comment (asomers) [reply src/unistd.rs]: What about this part?

Comment (F1rst-Unicorn) [reply src/unistd.rs]: Sorry, didn't read close enough. I've fixed it now.

**Pull request merged**